server/src/main/java/org/elasticsearch/node/NodeRoleSettings.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.node;
+
+import org.elasticsearch.cluster.node.DiscoveryNode;
+import org.elasticsearch.cluster.node.DiscoveryNodeRole;
+import org.elasticsearch.common.settings.Setting;
+import org.elasticsearch.common.settings.Setting.Property;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class NodeRoleSettings {
+
+    public static final Setting<List<DiscoveryNodeRole>> NODE_ROLES_SETTING = Setting.listSetting(
+        "node.roles",
```

Comment:
Would there be any benefit to extracting this into a public constant, so that other classes can import it, instead of rekeying it? I'm just imagining typos, or trying to find usages with an IDE. Just a thought.

Reply:
The typical way that we refer to these is via `NODE_ROLES_SETTING.getKey()`. This way, we can find all usages of the setting, whether it's the actual setting representation, or the key, in a single use of "find usages", rather than having to find usages for the key, and for the setting separately.

Reply:
Ah, that's what I ought to have said, having done exactly the same thing myself. 👍